core/types/log.go

```diff
@@ -44,11 +44,11 @@ type Log struct {
 	// hash of the transaction
 	TxHash common.Hash `json:"transactionHash" gencodec:"required"`
 	// index of the transaction in the block
-	TxIndex uint `json:"transactionIndex" gencodec:"required"`
```

Comment:
Any particular reason to drop this restriction?

Reply:
No block = no tx index.